src/main/java/seedu/address/model/EduMate.java

```diff
@@ -203,10 +207,23 @@ public User getUser() {
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof EduMate // instanceof handles nulls
-                && persons.equals(((EduMate) other).persons)
-                && user.equals(((EduMate) other).user));
+        if (this == other) {
+            return true;
+        }
+
+        if (!(other instanceof EduMate)) {
+            return false;
+        }
+
+        EduMate otherEduMate = (EduMate) other;
+
+        logger.info("Persons equality: "
+                + persons.equals(otherEduMate.persons));
+        logger.info("User equality: "
+                + user.equals(otherEduMate.user));
+
+        return persons.equals(otherEduMate.persons)
+                && user.equals(otherEduMate.user);
```

Comment:
Good to see you adapted the `equals` method to display the Happy Path more prominently. Also, good use of defensive programming strategies.